blockchain/vm/evm_test.go

```diff
@@ -0,0 +1,108 @@
+	// Condition 3. Caller Contract Deploy - after IstanbulCompatible Change, Call - after istanbulCompatible
+	{"0x3fd", vmLogInput, TestField{true, Block5}, ExpectField{200, vmLogOutput, nil}},
+	{"0x3fe", feePayerInput, TestField{true, Block5}, ExpectField{300, feePayerOutput, nil}},
+	{"0x3ff", validateInput, TestField{true, Block5}, ExpectField{5000, validateOutput, nil}},
+
+	// Expect Error Behavior
+	// Condition 1. Caller Contract Deploy - before IstanbulCompatible Change, Call - before istanbulCompatible
+	{"0x3fd", vmLogInput, TestField{false, Block4}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	{"0x3fe", feePayerInput, TestField{false, Block4}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	{"0x3ff", validateInput, TestField{false, Block4}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	// Condition 2. Caller Contract Deploy - before IstanbulCompatible Change, Call - after istanbulCompatible
+	{"0x3fd", vmLogInput, TestField{false, Block5}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	{"0x3fe", feePayerInput, TestField{false, Block5}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	{"0x3ff", validateInput, TestField{false, Block5}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	// Condition 2. Caller Contract Deploy - after IstanbulCompatible Change, Call - after istanbulCompatible
+	{"0x009", vmLogInput, TestField{true, Block5}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	{"0x00a", feePayerInput, TestField{true, Block5}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+	{"0x00b", validateInput, TestField{true, Block5}, ExpectField{0, "", kerrors.ErrPrecompiledContractAddress}},
+}
+
+func TestPrecompiledContractAddressMapping(t *testing.T) {
+	config := params.TestChainConfig        // Make ChainConfig for the test
+	config.IstanbulCompatibleBlock = Block5 // Set IstanbulCompatible block number as '5'
+
+	// Run Tests
+	for _, tt := range PrecompiledContractAddressMappingTestData {
+		// Make StateDB
+		callerAddr := common.BytesToAddress([]byte("contract"))
+		statedb, _ := state.New(common.Hash{}, state.NewDatabase(database.NewMemoryDBManager()))
+		format := params.CodeFormatEVM
+		format.SetIstanbulHFField(tt.isIstanbul)
+		statedb.CreateSmartContractAccount(callerAddr, format)
+
+		// Make EVM environment
+		vmctx := Context{
+			CanTransfer: func(StateDB, common.Address, *big.Int) bool { return true },
+			Transfer:    func(StateDB, common.Address, common.Address, *big.Int) {},
+			BlockNumber: tt.block,
+		}
+		vmenv := NewEVM(vmctx, statedb, config, &Config{})
+
+		// run
+		ret, gas, err := vmenv.Call(AccountRef(callerAddr), common.HexToAddress(tt.addr), tt.input, math.MaxUint64, new(big.Int))
+
+		// compare with expected data
+		assert.Equal(t, tt.ret, common.Bytes2Hex(ret))
+		assert.Equal(t, tt.gas, math.MaxUint64-gas)
+		assert.Equal(t, tt.err, err)
```

Comment:
I think you declared `TestField` and `ExpectedField` to categorize related values. However, if you do not give them  names, it is hard to understand where these fields come from. It is a matter of taste, but I think explicit naming helps readers understand the structure of the code, though we should type a bit more codes. 😅

Reply:
Thanks for your advice, please take a look https://github.com/klaytn/klaytn/pull/912/commits/b62138adad3b89433c763458df317a578be8d84a